presentation-compiler/src/main/dotty/tools/pc/SelectionRangeProvider.scala

```diff
@@ -94,16 +97,30 @@ class SelectionRangeProvider(driver: InteractiveDriver, params: ju.List[OffsetPa
           val srcPos = list.head.sourcePos
           val lastSpan = list.last.span
           val allArgsSrcPos = SourcePosition(srcPos.source, srcPos.span union lastSpan, srcPos.outer)
-          if allArgsSrcPos.contains(pos) then Some(toSelectionRange(allArgsSrcPos))
-          else None
-
-    tree match
-      case DefDef(_, paramss, _, _) => paramss.flatMap(allArgsSelectionRange) ++ treeSelectionRange
-      case Apply(_, args) => allArgsSelectionRange(args) ++ treeSelectionRange
-      case TypeApply(_, args) => allArgsSelectionRange(args) ++ treeSelectionRange
-      case UnApply(_, _, pattern) => allArgsSelectionRange(pattern) ++ treeSelectionRange
-      case Function(args, body) => allArgsSelectionRange(args) ++ treeSelectionRange
-      case _ => treeSelectionRange
+          maybeToSelectionRange(allArgsSrcPos)
+
+    val allSelectionRanges: Iterable[SelectionRange] = tree match
+      case vdef @ ValDef(_, tpt, _) =>
+        maybeToSelectionRange(vdef.namePos)
+        ++ maybeToSelectionRange(tpt.sourcePos)
+        ++ maybeToSelectionRange(vdef.rhs.sourcePos)
```

Comment:
Why do we need rhs? Wouldn't that already be added by the tree inside RHS? Same for tpt. untpdPath should give us all tree nodes on the way to the current position.